lib/Doctrine/DBAL/Driver/AbstractSQLServerDriver/Exception/PortWithoutHost.php

```diff
@@ -2,16 +2,16 @@
 
 declare(strict_types=1);
 
-namespace Doctrine\DBAL\Driver\AbstractSQLServerDriver;
+namespace Doctrine\DBAL\Driver\AbstractSQLServerDriver\Exception;
 
-use Doctrine\DBAL\Driver\AbstractDriverException;
+use Doctrine\DBAL\Driver\AbstractException;
 
 /**
  * @internal
  *
  * @psalm-immutable
  */
-final class PortWithoutHost extends AbstractDriverException
+final class PortWithoutHost extends AbstractException
```

Comment:
this is a BC break, because catching `AbstractDriverException` will not catch this exception anymore (remember that *using* the exception classes is not only done in instantiations but also in `catch` clauses)

Reply:
The class is marked as `internal` though.

Reply:
@greg0ire is right. Consumers should be only catching `Driver\DriverException`.

Reply:
Ah but AbstractDriverException itself is not marked as `internal`, so they might be, might they not?

Reply:
I see. Let's revert this line. It looks like the only change that needs to be reverted. The rest of the exceptions changed in a similar way were introduced in 2.11.x.

Reply:
So this might be something that the BackwardCompatibilityCheck doesn't yet detect: even if the class is internal if the new version is not a subtype of the old version, it's a BC break. Even if it's not an exception.

Reply:
well, for internal classes which are not an exception, that's fine (as it is not covered by the BC promise). For exception, that's a bit more complex: due to the catching semantic, even an internal exception class must still stay a subtype of all its non-internal types of the old version.

Reply:
How is catching an exception different from accepting an argument of a given type by a function?

Reply:
hmm, indeed. So caring about non-internal parents of internal classes indeed applies to other cases too.

Reply:
Reported as https://github.com/Roave/BackwardCompatibilityCheck/issues/256.